src/InputForm.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+
+interface InputFormType {
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setOnEmptyFormSubmit: React.Dispatch<React.SetStateAction<boolean>>,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>
+  isInputDisabled: boolean;
+  setIsInputDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const InputForm: React.FC<InputFormType>
+  = ({
+    setFilteredTodos,
+    setOnEmptyFormSubmit,
+    setTempTodo,
+    isInputDisabled,
+    setIsInputDisabled,
+  }) => {
+    const [inputValue, setInputValue] = useState('');
+
+    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setInputValue(event.target.value);
+    };
+
+    const handleFormSubmit = async (event:
+    React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      if (!inputValue) {
+        setOnEmptyFormSubmit(true);
+
+        return;
+      }
+
+      const todoData = {
+        id: 0,
```

Comment:
The todo id is hardcoded to 0. This could potentially cause conflicts if todos with the same id are added. Consider using a unique id generator for each todo.